mypy/partially_defined.py

```diff
@@ -407,7 +438,7 @@ def visit_name_expr(self, o: NameExpr) -> None:
             self.tracker.record_definition(o.name)
         elif self.tracker.is_defined_in_different_branch(o.name):
             # A variable is defined in one branch but used in a different branch.
-            if self.loop_depth > 0:
+            if len(self.loops) > 0:
```

Comment:
You can write simply like this, empty lists are falsey.
```suggestion
            if self.loops:
```